test/test_settings.py

```diff
@@ -367,104 +367,90 @@ class TestSettingsEquality(object):
-            h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS: 4,
-            h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: 2**16,
-        }
-        return h2.settings.Settings(client=True, initial_values=overrides)
-
-    def another_instance(self):
-        """
-        Return an instance of the class under test.  Each call to this method
-        must return a different object.  The objects must not be equal to the
-        objects returned by an_instance.  They may or may not be equal to
-        each other (they will not be compared against each other).
-        """
-        overrides = {
-            h2.settings.SettingCodes.HEADER_TABLE_SIZE: 8080,
-            h2.settings.SettingCodes.MAX_FRAME_SIZE: 16388,
-            h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS: 100,
-            h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: 2**16,
-        }
-        return h2.settings.Settings(client=False, initial_values=overrides)
-
-    def test_identical_eq(self):
+    SettingsStrategy = strategies.builds(
+        h2.settings.Settings,
+        client=strategies.booleans(),
+        initial_values=strategies.dictionaries(
+            keys=strategies.sampled_from([
+                h2.settings.SettingCodes.HEADER_TABLE_SIZE,
+                h2.settings.SettingCodes.ENABLE_PUSH,
+                h2.settings.SettingCodes.INITIAL_WINDOW_SIZE,
+                h2.settings.SettingCodes.MAX_FRAME_SIZE,
+                h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS,
+                h2.settings.SettingCodes.MAX_FRAME_SIZE,
+                h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE,
+            ]),
+            values=strategies.integers(min_value=0))
+        ) 
+
+    @given(o=SettingsStrategy)
+    def test_identical_eq(self, o):
         """
         An object compares equal to itself using the == operator.
         """
-        o = self.an_instance()
         assert (o == o)
 
-    def test_identical_ne(self):
+    @given(o=SettingsStrategy)
+    def test_identical_ne(self, o):
         """
         An object doesn't compare not equal to itself using the != operator.
         """
-        o = self.an_instance()
         assert not (o != o)
 
-    def test_same_eq(self):
+    @given(a=SettingsStrategy)
+    def test_same_eq(self, a):
         """
         Two objects that are equal to each other compare equal to each other
         using the == operator.
         """
-        a = self.an_instance()
-        b = self.an_instance()
+        b = copy.deepcopy(a)
         assert (a == b)
 
-    def test_same_ne(self):
+    @given(a=SettingsStrategy)
+    def test_same_ne(self, a):
         """
         Two objects that are equal to each other do not compare not equal to
         each other using the != operator.
         """
-        a = self.an_instance()
-        b = self.an_instance()
+        b = copy.deepcopy(a)
         assert not (a != b)
 
-    def test_different_eq(self):
+    @given(a=SettingsStrategy, b=SettingsStrategy)
```

Comment:
So it's entirely possible that this would generate two settings objects that *are* identical. We need to be a bit careful here and construct a test that allows us to ensure that the two objects are different. In this case, the easiest way might be to create two strategies that cannot generate overlapping Settings objects in some way, or at least create a Settings object that our strategy cannot generate. Of course, if @alexwlchan has more useful things to suggest here that'd be really helpful (he's our resident Hypothesis expert).